contracts/optimism/TokenRateOracle.sol

```diff
@@ -95,15 +110,66 @@ contract TokenRateOracle is CrossDomainEnabled, ITokenRateOracle, Versioned {
         MAX_ALLOWED_TOKEN_RATE_DEVIATION_PER_DAY = maxAllowedTokenRateDeviationPerDay_;
     }
 
-    function initialize(uint256 tokenRate_, uint256 rateL1Timestamp_) external {
+    /// @notice Initializes the contract from scratch.
+    /// @param admin_ Address of the account to grant the DEFAULT_ADMIN_ROLE
+    /// @param tokenRate_ wstETH/stETH token rate.
+    /// @param rateUpdateL1Timestamp_ L1 time when rate was updated on L1 side.
+    function initialize(address admin_, uint256 tokenRate_, uint256 rateUpdateL1Timestamp_) external {
+        if (admin_ == address(0)) {
+            revert ErrorZeroAddressAdmin();
+        }
         if (tokenRate_ < MIN_ALLOWED_TOKEN_RATE || tokenRate_ > MAX_ALLOWED_TOKEN_RATE) {
             revert ErrorTokenRateInitializationIsOutOfAllowedRange(tokenRate_);
         }
-        if (rateL1Timestamp_ > block.timestamp + MAX_ALLOWED_L2_TO_L1_CLOCK_LAG) {
-            revert ErrorL1TimestampInitializationIsOutOfAllowedRange(rateL1Timestamp_);
+        if (rateUpdateL1Timestamp_ > block.timestamp + MAX_ALLOWED_L2_TO_L1_CLOCK_LAG) {
+            revert ErrorL1TimestampInitializationIsOutOfAllowedRange(rateUpdateL1Timestamp_);
         }
         _initializeContractVersionTo(1);
-        _setTokenRateAndTimestamps(uint128(tokenRate_), uint64(rateL1Timestamp_));
+        _grantRole(DEFAULT_ADMIN_ROLE, admin_);
+        _addTokenRate(uint128(tokenRate_), uint64(rateUpdateL1Timestamp_), uint64(block.timestamp));
+    }
+
+    /// @notice Pauses token rate updates and setup old rate provided by tokenRateIndex_.
+    ///         Should be called by DAO or multisig only.
+    /// @param tokenRateIndex_ The index of the token rate that applies after the pause.
+    ///        Token Rate can't be received older then MAX_DELTA_TIME_TO_PAUSE_TOKEN_RATE_UPDATES.
+    function pauseTokenRateUpdates(uint256 tokenRateIndex_) external onlyRole(RATE_UPDATE_DISABLER_ROLE) {
```

Comment:
sorry, this implementation doesn't meet the following expectations:
- if `tokenRateIndex_` is the last element, than it mustn't check the `MAX_DELTA_TIME_TO_PAUSE_TOKEN_RATE_UPDATES`
- anyways, l2 reception timestamp should be rolled back to make the rate outdated **immediately**

Reply:
Re(2): not sure I've got it, but isn't it enough to always return `false` in `isLikelyOutdated` if `isPaused`?

Reply:
ReRe(1): it has been implemented after the comment I posted, yeah :+1: 
ReRe(2): good idea by the way!